### PR TITLE
Updated docs for pullback.

### PR DESCRIPTION
**Pull request by mbrandonw**
Beefing up the docs for pullback.

### CODE REVIEW

File: Sources/SnapshotTesting/Snapshotting.swift [hunk 1]
```diff
@@ -46,8 +46,44 @@ public struct Snapshotting<Value, Format> {
...
+  /// - Parameters:
+  ///   - transform: A transform function from `NewValue` into `Value`.
+  ///   - otherValue: A value to be transformed.
+  public func pullback<NewValue>(_ transform: @escaping (_ otherValue: NewValue) -> Value) -> Snapshotting<NewValue, Format> {
+    return self.asyncPullback { newValue in Async(value: transform(newValue)) }
+  }
+
   /// Transforms a strategy on `Value`s into a strategy on `NewValue`s through a function `(NewValue) -> Async<Value>`.
   ///
+  /// See the documention of `pullback` for a full description of how pullbacks works. This operation differs
```
Comment (stephencelis) [inline Sources/SnapshotTesting/Snapshotting.swift]: Should we try a [`seeAlso:`](https://developer.apple.com/library/archive/documentation/Xcode/Reference/xcode_markup_formatting_ref/SeeAlso.html) callout?

Comment (mbrandonw) [reply Sources/SnapshotTesting/Snapshotting.swift]: ah yes, lemme try it out

File: Sources/SnapshotTesting/Snapshotting.swift [hunk 2]
```diff
@@ -46,8 +46,44 @@ public struct Snapshotting<Value, Format> {
     }
   }
 
+  /// Transforms a strategy on `Value`s into a strategy on `NewValue`s through a function
+  /// `(NewValue) -> Value`.
+  ///
+  /// This is the most important operation for transforming existing strategies into new strategies. It
```
Comment (stephencelis) [inline Sources/SnapshotTesting/Snapshotting.swift]: For docs I've been letting Xcode wrap paragraphs on single lines. Just seems annoying to have to manage the line breaks of prose as it gets updated. Down to pick either lane, though. We should just be consistent.

File: Sources/SnapshotTesting/Snapshotting.swift [hunk 3]
```diff
@@ -46,8 +46,44 @@ public struct Snapshotting<Value, Format> {
...
+  /// A simple example of this is to `pullback` the snapshot strategy on `UIView`'s to work on
+  /// `UIViewController`'s:
+  ///
+  /// ```
+  /// let strategy = Snapshotting<UIView, UIImage>.image.pullback { (vc: UIViewController) in
+  ///   return vc.view
+  /// }
+  /// ```
+  ///
+  /// Here we took the strategy that snapshots `UIView`'s as `UIImage`'s and pulled it back to work on
```
Comment (stephencelis) [inline Sources/SnapshotTesting/Snapshotting.swift]:
```suggestion
  /// Here we took the strategy that snapshots `UIView`s as `UIImage`s and pulled it back to work on
```

File: Sources/SnapshotTesting/Snapshotting.swift [hunk 4]
```diff
@@ -46,8 +46,44 @@ public struct Snapshotting<Value, Format> {
...
+  /// `(NewValue) -> Value`.
+  ///
+  /// This is the most important operation for transforming existing strategies into new strategies. It
+  /// allows you to transform a `Snapshotting<Value, Format>` into a `Snapshotting<NewValue, Format>`
+  /// by pulling it back along a function `(NewValue) -> Value`. Notice that the function must go in
+  /// in the direction `(NewValue) -> Value` even though we are transforming in the other direction
+  /// `(Snapshotting<Value, Format>) -> Snapshotting<NewValue, Format>`.
+  ///
+  /// A simple example of this is to `pullback` the snapshot strategy on `UIView`'s to work on
+  /// `UIViewController`'s:
```
Comment (stephencelis) [inline Sources/SnapshotTesting/Snapshotting.swift]:
```suggestion
  /// `UIViewController`s:
```

File: Sources/SnapshotTesting/Snapshotting.swift [hunk 5]
```diff
@@ -46,8 +46,44 @@ public struct Snapshotting<Value, Format> {
...
+  /// Transforms a strategy on `Value`s into a strategy on `NewValue`s through a function
+  /// `(NewValue) -> Value`.
+  ///
+  /// This is the most important operation for transforming existing strategies into new strategies. It
+  /// allows you to transform a `Snapshotting<Value, Format>` into a `Snapshotting<NewValue, Format>`
+  /// by pulling it back along a function `(NewValue) -> Value`. Notice that the function must go in
+  /// in the direction `(NewValue) -> Value` even though we are transforming in the other direction
+  /// `(Snapshotting<Value, Format>) -> Snapshotting<NewValue, Format>`.
+  ///
+  /// A simple example of this is to `pullback` the snapshot strategy on `UIView`'s to work on
```
Comment (stephencelis) [inline Sources/SnapshotTesting/Snapshotting.swift]:
```suggestion
  /// A simple example of this is to `pullback` the snapshot strategy on `UIView`s to work on
```

File: Sources/SnapshotTesting/Snapshotting.swift [hunk 6]
```diff
@@ -46,8 +46,44 @@ public struct Snapshotting<Value, Format> {
...
+  /// `UIViewController`'s:
+  ///
+  /// ```
+  /// let strategy = Snapshotting<UIView, UIImage>.image.pullback { (vc: UIViewController) in
```
Comment (stephencelis) [inline Sources/SnapshotTesting/Snapshotting.swift]: I forget if I've been doing 4-space indentation or triple-backticks in the docs. Down for either, just think we should be consistent.

File: Sources/SnapshotTesting/Snapshotting.swift [hunk 7]
```diff
@@ -46,8 +46,44 @@ public struct Snapshotting<Value, Format> {
...
+  ///   return vc.view
+  /// }
+  /// ```
+  ///
+  /// Here we took the strategy that snapshots `UIView`'s as `UIImage`'s and pulled it back to work on
+  /// `UIViewController`'s by using the function `(UIViewController) -> UIView` that simply plucks the
```
Comment (stephencelis) [inline Sources/SnapshotTesting/Snapshotting.swift]:
```suggestion
  /// `UIViewController`s by using the function `(UIViewController) -> UIView` that simply plucks the
```

File: Sources/SnapshotTesting/Snapshotting.swift [hunk 8]
```diff
@@ -46,8 +46,44 @@ public struct Snapshotting<Value, Format> {
...
+  /// view out of the controller.
+  ///
+  /// Nearly every snapshot strategy provided in this library is a `pullback` of some base strategy, which
```
Comment (stephencelis) [inline Sources/SnapshotTesting/Snapshotting.swift]:
```suggestion
  /// Nearly every snapshot strategy provided in this library is a pullback of some base strategy, which
```

File: Sources/SnapshotTesting/Snapshotting.swift (hunk 2 above)
Comment (mbrandonw) [reply Sources/SnapshotTesting/Snapshotting.swift]: I just like it for the PR readability. also if we were to ever lint our oss (which might be necessary with contributors) I image we would want a line length rule.

File: Sources/SnapshotTesting/Snapshotting.swift (hunk 6 above)
Comment (mbrandonw) [reply Sources/SnapshotTesting/Snapshotting.swift]: ticks seem easiest to do.

**Comment by mbrandonw** [comment]
I updated the doc style match what is in the repo, so gonna merge.

**Pull request merged**